File: 3. Arrays and String/485. Max Consecutive Ones/max_consecutive.cpp
class Solution
{
public:
    int findMaxConsecutiveOnes(vector<int> &nums)
    {
        int max = 0;
        int curr = 0;

        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] == 1)
            {
                curr += 1;
            }
            else
            {
                max = curr > max ? curr : max;
                curr = 0;
            }
        }
        max = curr > max ? curr : max;

        return max;
    }
};

File: 3. Arrays and String/485. Max Consecutive Ones/max_consecutive.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        maximum = 0
        curr = 0

        for i in range(len(nums)):
            if nums[i] == 1:
                curr += 1
            else:
                maximum = maximum if maximum > curr else curr
                curr = 0
        maximum = maximum if maximum > curr else curr
        return maximum